include/quic/network.hpp

```diff
@@ -12,172 +12,96 @@ extern "C"
-        // passed:
-        //
-        //      local_addr                          OPTIONAL (if not, the "any" address will be used)
-        //      {
-        //          std::string host,
-        //          uint16_t port
-        //      }
-        //      or local_addr{uint16_t port} for any address with the given port.
-        //
-        //      remote_addr                         REQUIRED
-        //      {
-        //          std::string host,
-        //          std::string port
-        //      }
-        //      client_tls                          REQUIRED
-        //      {
-        //          std::string client_key,         OPTIONAL (required if using client certificate
-        //          std::string client_cert             authentication by server)
-        //
-        //          std::string server_cert         (A) REQUIRED (pick ***one*** of options A/B/C)
-        //          std::string server_CA           (B)
-        //      }
-        //      client_tls_callback_t client_tls_cb (C)
-        //
-        template <typename... Opt>
-        std::shared_ptr<Client> client_connect(Opt&&... opts)
-        {
-            std::promise<std::shared_ptr<Client>> p;
-            auto f = p.get_future();
-            quic_manager->call([&opts..., &p, this]() mutable {
-                try
-                {
-                    // initialize client context and client tls context simultaneously
-                    std::shared_ptr<ClientContext> client_ctx =
-                            std::make_shared<ClientContext>(quic_manager, std::forward<Opt>(opts)...);
-
-                    client_ctx->udp_handle = handle_mapping(false, client_ctx->local);
-
-                    // ensure addresses stored correctly
-                    log::trace(log_cat, "Client local addr: {}", client_ctx->local);
-                    log::trace(log_cat, "Client remote addr: {}", client_ctx->remote);
-
-                    // create client and then copy assign it to the client context so we can return
-                    // the shared ptr from this function
-                    auto client_ptr = std::make_shared<Client>(
-                            quic_manager, client_ctx, (*client_ctx).conn_id, client_ctx->udp_handle);
-                    client_ctx->client = client_ptr;
-
-                    quic_manager->clients.emplace_back(std::move(client_ctx));
-                    log::trace(log_cat, "Client context emplaced");
-
-                    p.set_value(client_ptr);
-                }
-                catch (...)
-                {
-                    p.set_exception(std::current_exception());
-                }
-            });
-
-            return f.get();
-        }
+        // Maps local listening address to respective endpoint
+        std::map<Address, std::shared_ptr<Endpoint>, addr_comparator> endpoint_map;
+        std::map<Address, std::shared_ptr<uv_udp_t>, addr_comparator> handle_map;
```

Comment:
This feels out of place for a `map` here as `Address` isn't something that is inherently orderable (and the order applied here is applying dictionary ordering on the stringified form of the address).

Is there a reason to avoid using an `unordered_map` here?